common/lib/capa/capa/correctmap.py

```diff
@@ -91,8 +91,11 @@ def set_dict(self, correct_map):
         # empty current dict
         self.__init__()
 
+        if not correct_map:
```

Comment:
In what situations would `correct_map` be invalid?

Reply:
Most probably for situations unknown and edge cases. There has already been that check which was preventing update of the `state` incase the correct map is invalid. I just abstracted it out and applied on both `if` and `else`. 

Currently, if the correct_map is invalid, it tries to itratre over it in `else` condition.

Reply:
Ah, okay. Thank you for explaining.